### PR TITLE
Add IsChainDigraph

### PR DESCRIPTION
**Pull request by AshleyClayton**
Adds a method "IsChainDigraph" to prop, with related documentation and tests.

### CODE REVIEW

File: gap/prop.gi [hunk 1]
```diff
@@ -8,6 +8,17 @@
...
 ##
 
+InstallMethod(IsChainDigraph, "for a digraph", [IsDigraph],
+function(digraph)
+  if DigraphNrEdges(digraph) = 0 and DigraphNrVertices(digraph) = 1 then
+    return true;
+  else
+    return IsConnectedDigraph(digraph) and InDegreeSet(digraph) = [0, 1]
+           and OutDegreeSet(digraph) = [0, 1]
+           and IsCycleDigraph(digraph) = false;
```
Comment (james-d-mitchell) [inline gap/prop.gi]: This could be `not IsCycleDigraph(digraph)` instead of `IsCycleDigraph(digraph) = false`.

File: doc/prop.xml [hunk 2]
```diff
@@ -8,6 +8,36 @@
...
+  <Prop Name="IsChainDigraph" Arg="digraph"/>
+  <Returns><K>true</K> or <K>false</K>.</Returns>
+  <Description>
+    <C>IsChainDigraph</C> returns <K>true</K> if the digraph <A>digraph</A> is
+    isomorphic to the chain digraph with the same number of vertices as
+    <A>digraph</A>, and <K>false</K> if it is not; see <Ref
+      Oper="ChainDigraph"/>.<P/>
+
+    A digraph is a <E>chain</E> if and only if it is connected,
+    not a cycle and both the in and out degree of any vertex is at most one.
```
Comment (james-d-mitchell) [inline doc/prop.xml]: So, in the method for `IsChainDigraph` you require `InDegreeSet(digraph) = [0, 1]` and `OutDegreeSet(digraph) = [0, 1]` which is not exactly the same as what you've written in the documentation. Can you correct this please?

Comment (AshleyClayton) [reply doc/prop.xml]: Thanks for the suggestion! In the special case of `ChainDigraph(1)`, the in and out degree sets are `[0]`, but I think that still makes my documentation statement incorrect, so I will rewrite that to clarify that case :)

File: gap/prop.gi [hunk 3]
```diff
@@ -8,6 +8,17 @@
 #############################################################################
 ##
 
+InstallMethod(IsChainDigraph, "for a digraph", [IsDigraph],
+function(digraph)
+  if DigraphNrEdges(digraph) = 0 and DigraphNrVertices(digraph) = 1 then
```
Comment (wilfwilson) [inline gap/prop.gi]: You could replace this whole function by

```
return IsDirectedTree(digraph) and IsSubset([0, 1], OutDegreeSet(digraph));
```

**Pull request merged**